src/docker.test.ts

```diff
@@ -117,14 +119,19 @@ describe("Docker images", (): void => {
     if (!cacheHit) {
       core.getInput.mockReturnValueOnce(readOnly.toString());
       if (!readOnly) {
-        core.getState.mockReturnValueOnce(preexistingImages.join("\n"));
-        const images = preexistingImages.concat(newImages);
-        util.execBashCommand.mockResolvedValueOnce(images.join("\n"));
+        if (prevSave) {
+          cache.restoreCache.mockResolvedValueOnce(key);
+        } else {
+          cache.restoreCache.mockResolvedValueOnce(key + "make keys mismatch");
```

Comment:
This is not a very realistic case to consider since `restoreCache` returns `undefined` when the cache isn't present.

Nit (moot): prefer a more natural-looking string that has a space after `key`. You may find it easier to remember this if you use string interpolation rather than string concatenation.